### ISSUE TITLE
Core miner

### ISSUE THREAD

**Issue by TwinNumberOne** [siggame/MegaMinerAI-Game-Ideas#63]
**TLDR**
-
- This idea is for multi miner
- the resources get richer as you go down and harder to mine
- You need to support the earth or you will get buried in rubble
- you can upgrade your miner's mining speed, move speed, hp, carry capacity
- It is faster to move through a mined area then to mine your way through
- victory by destroying the enemy base or mining the most resources

**Story**
-
In this game you are one of two rival corporations both won the right to mine a valuable section of land. Your goal is to mine as many resources as possible or to destroy your enemies base to stop them from being able to mine the land. Due to how it gets harder to mine the deeper you go you will need some upgraded drills to mine it, and you will need to specialize to make sure that you can mine efficiently.

**The Game**
-
**Victory**
You can win in one to two ways mine the most resources when the timer runs out, or destroy the enemy base. This means the total amount of resources you have mined meaning even if you spend the resources it will count towards your score. The base is destroyed by mining the blocks under it making the land above unstabled (due mainly due to the base being on the blocks above) or by mining into the base itself.

**Map**
- a 120x120 with the top layer being open-air with two bases at opposite sides.
- There are 10 layers where every layer the block hp increases by 10 but also becomes richer in resources
- Resources will be distributed in ore veins around the map, with the rest being earth.
- All blocks can be turned into tunnels even ore.
- Tunneling and mining affect how stable the earth is, where a tunnel can support more weight then blocks with an open block below them. Where the support item keeps the earth stable.
- If you mine a tunnel through the block you only collect half the resources unless you mine it more, but doing so destabilizes the tunnel which can lead to collapse. 
- Gravel made by blocks falling due to being destabilized always falls and will burry miners causing damage
- The base can be destroyed by mining the ground from under it or being damaged directly

**Movement**
- the base units can only move 1 tile by base, 10 at the max movement upgrade. 
- When a unit finishes mining a block (bringing its hp to 0, or by making it a tunnel) it can move into that block that same tick.
- it must deal half a blocks hp to it to mine out a tunnel. (this is a special action it must take, and yes they can mine tunnels)
- should a unit be mining a block and a unit moves into that space (either by turning it into a tunnel) it will take the damage that the block would have taken
- A unit can dump its inventory to backfill an area (the block gaining more hp the more they dump)
- A unit when digging through unsupported gravel it will not form into a tunnel unless supports are used (meaning it won't move into the block first)
- if a units tries to dig a tunnel it will the moves into the block and be buried where its damage output will be reduced by half. Another unit can dig it out but the unit being dug out will take damage.
- A unit will take damage if it falls 2 blocks (increasing damage the further it falls), and it cannot be more than 1 block above a solid block (meaning it needs supports to move directly up).

**Resource collection**
- if you mine a tunnel you will only get half the resource value of the block
- you get a portion of the blocks resources depending on how much damage you dealt to it
- when you mine earth you collect dirt, and if you mine ore you get ore. with the amount being equal to the blocks hp (meaning a 50hp ore block has 50 ore)
- resources are able to be dropped into the base when the miner is adjacent to the base.

**Unit**
Each units by base does 5 damage when it mines, moves 1 block, can hold 50 items worth of material, has a hp of 10, all of which are upgradable. All upgrades are at lvl 0.
Each turn a unit can mine, and do either an action (planting a bomb, installing a support) or move.
Units can also spend a turn to give another unit resources (aka trade inventory).

**Upgrades**

| Ability | Cost per level | increases by |  max value |
|----|----|----|----|
|**Health**          |10|5|200|
|**Storage**             |20|10|500|
|**Movement**           |10|1|10|
|**Drill**              |10|5|100|

**Useable**
The destructive usables are able to be trigger by the AI whenever they want (even in the storage on a unit, doing that much damage to the unit), while useable like shielding and supports can be built in an adjacent clear block. All hoppers must be connected to the base by another hopper or be touching the base, these are a modification to a block and are destroyed if the block is damaged in any way. Should a block be destroyed by a TNT or a blast charge all the resources of that block are destroyed. Support are destroyed by any drilling or bombs. As you can expect from such a cheap structure.

| Name | Cost | what it does |
|----|----|----|
|**Support**          |5|makes blocks not collapse, and makes movement faster|
|**TNT**             |10|does 5 damage to blocks in a 3x3, stacks|
|**Blash charge**           |10|does 10 damage to 2 blocks above, stacks|
|**sheilding**              |5|increases a blocks hp by 5 per instalation|
|**hopper**              |50|extends where a miner can put resources into the base|

**Combat**
Should another miner be in a block another miner is mining it will take damage equal to that of which the drill of the miner can output.

**Remarks**
This game is at least supposed to be a simple mining game, where combat is an option. The idea is that the complexity comes from how the players decide to make their routes, and how they decide to use the same structures to work together to increase their mining efficiency, as well as dealing with how resources are distributed. Also playing with gravel could be fun, but to make it easier on the AI I would see about giving them a tool that tells them the integrity of a bock before and after the mine. Also, I have an idea that a block can support 3 times is hp in blocks above if it has two blocks to is left and right, but it can only support 2x its weight in blocks if it is missing one of those blocks, but that might be making it too complicated, and just calculating how much block hp it can support based on its hp.  But if dealing with gravel is too much work you can make it so supports give a speed buff, that would be fine too.
